userspace/libsinsp/threadinfo.h

```diff
@@ -682,6 +658,65 @@ class SINSP_PUBLIC sinsp_thread_manager
 
 	void set_m_max_n_proc_lookups(int32_t val) { m_max_n_proc_lookups = val; }
 	void set_m_max_n_proc_socket_lookups(int32_t val) { m_max_n_proc_socket_lookups = val; }
+
+	// ---- libsinsp::state::table implementation ----
+
+	size_t entries_count() const override
+	{
+		return m_threadtable.size();
+	}
+
+	void clear_entries() override
+	{
+		m_threadtable.clear();
+	}
+
+	std::unique_ptr<libsinsp::state::table_entry> new_entry() const override;
+
+	bool foreach_entry(std::function<bool(libsinsp::state::table_entry& e)> pred) override
+	{
+		return m_threadtable.loop([&pred](sinsp_threadinfo& e){ return pred(e); });
+	}
+
+	std::shared_ptr<libsinsp::state::table_entry> get_entry(const int64_t& key) override
+	{
+		return find_thread(key, false);
+	}
+
+	std::shared_ptr<libsinsp::state::table_entry> add_entry(const int64_t& key, std::unique_ptr<libsinsp::state::table_entry> entry) override
+	{
+		if (!entry)
+		{
+			throw sinsp_exception("null entry added to thread table");
+		}
+		auto tinfo = dynamic_cast<sinsp_threadinfo*>(entry.get());
+		if (!tinfo)
+		{
+			throw sinsp_exception("unknown entry type added to thread table");
+		}
+		if (tinfo->m_tid != key)
+		{
+			throw sinsp_exception("key does not match pid of entry added to thread table");
+		}
+		entry.release();
+		add_thread(tinfo, false);
+		return get_entry(key);
+	}
+
+	bool erase_entry(const int64_t& key) override
+	{
+		// todo(jasondellaluce): should we trigger the whole removal logic,
```

Comment:
True, however this could become dangerous if we remove a thread in the middle of two `sinsp::next` invocations. I think this is the purpose of `remove_thread`, and I'm not sure we'd want to diverge from that for now. A potential alternative would be to not support the `erase_entry` method for now and make it throw a "non supported" exception for v1 of this feature.

Reply:
we can keep it as it is right now, we just need to remember that a call to `erase_entry`  doesn't always correspond to a real thread deletion in the table